### PR TITLE
Add ruby support

### LINKED ISSUE #114: Ruby runtime
Just opening an issue to track/suggest a ruby shim as well, I may take a crack at building one if I have time, looks like it may be pretty simple, possibly also worth making a ruby gem to standardize some interfaces for operating with whatever minimal/standard apex ruby shim gets merged in here.

### PR DESCRIPTION
**Pull request by minamijoyo**
Fixes #114

https://aws.amazon.com/about-aws/whats-new/2018/11/aws-lambda-supports-ruby/

```
$ cd _examples/ruby

$ $GOPATH/bin/apex deploy --dry-run


  + function ruby_simple
    runtime: ruby2.5
    memory: 128
    timeout: 3
    handler: lambda.handler

  + alias ruby_simple
    version: 1
    alias: current

  + function ruby_dependency
    runtime: ruby2.5
    memory: 128
    timeout: 3
    handler: lambda.handler

  + alias ruby_dependency
    alias: current
    version: 1

$ $GOPATH/bin/apex deploy
   • creating function         env= function=simple
   • creating function         env= function=dependency
   • created alias current     env= function=simple version=3
   • function created          env= function=simple name=ruby_simple version=3
   • created alias current     env= function=dependency version=3
   • function created          env= function=dependency name=ruby_dependency version=3
```

```
$ $GOPATH/bin/apex invoke simple < event.json
null

$ $GOPATH/bin/apex logs simple --since 5m
/aws/lambda/ruby_simple start simple function
/aws/lambda/ruby_simple START RequestId: dcac4cb2-f46a-11e8-9186-834116da1e63 Version: 3
/aws/lambda/ruby_simple processing event: {"hello"=>"world"}
/aws/lambda/ruby_simple END RequestId: dcac4cb2-f46a-11e8-9186-834116da1e63
/aws/lambda/ruby_simple REPORT RequestId: dcac4cb2-f46a-11e8-9186-834116da1e63  Duration: 7.80 ms       Billed Duration: 100 ms       Memory Size: 128 MB     Max Memory Used: 18 MB
```

```
$ $GOPATH/bin/apex invoke dependency < event.json

$ $GOPATH/bin/apex logs dependency --since 5m
/aws/lambda/ruby_dependency start dependency function
/aws/lambda/ruby_dependency START RequestId: f2e98178-f46a-11e8-acbb-910b7838e0fd Version: 3
/aws/lambda/ruby_dependency event.blank?: false
/aws/lambda/ruby_dependency END RequestId: f2e98178-f46a-11e8-acbb-910b7838e0fd
/aws/lambda/ruby_dependency REPORT RequestId: f2e98178-f46a-11e8-acbb-910b7838e0fd      Duration: 2.46 ms       Billed Duration: 100 ms       Memory Size: 128 MB     Max Memory Used: 19 MB
```

### CODE REVIEW

**Comment by minamijoyo** [comment]
@tj I'd appreciate it if you could review this.

**Comment by tj** [comment]
LGTM! I might not be able to cut a release soon though, I remember I ran into some weird build issues last time, haven't had time to look into it

File: plugins/inference/inference.go [hunk 1]
```diff
@@ -20,6 +21,7 @@ func init() {
 			"main.go":             golang.Runtime,
 			"target/apex.jar":     java.Runtime,
 			"build/libs/apex.jar": java.Runtime,
+			"lambda.rb":           ruby.Runtime,
```
Comment (tj) [inline plugins/inference/inference.go]: wonder if we should change this to main.rb, unless there's an idiomatic "entrypoint" file name for Ruby stuff (like index.js)

Comment (minamijoyo) [reply plugins/inference/inference.go]: @tj Thank you for reviewing and merging this 😄 
Now, this pull request has been already merged, so I've just opened a new issue #922 to discuss a filename.

**Pull request merged**